### ISSUE TITLE
Intrinsic failures

### ISSUE THREAD

**Issue by philberty** [Rust-GCC/gccrs#1674]
I tried this code:

```rust
Error: some tests are not correct
Error: -O1   scan-tree-dump-times original "(?n)f32 = __builtin_sqrtf \\(5\\.0e\\+0\\);$" 1
Error: -O1   scan-tree-dump-times original "(?n)f32 = __builtin_sinf \\(3\\.9e\\+1\\);$" 1
Error: -O2   scan-tree-dump-times original "(?n)f32 = __builtin_sqrtf \\(5\\.0e\\+0\\);$" 1
Error: -O2   scan-tree-dump-times original "(?n)f32 = __builtin_sinf \\(3\\.9e\\+1\\);$" 1
Error: -O3 -g   scan-tree-dump-times original "(?n)f32 = __builtin_sqrtf \\(5\\.0e\\+0\\);$" 1
Error: -O3 -g   scan-tree-dump-times original "(?n)f32 = __builtin_sinf \\(3\\.9e\\+1\\);$" 1
Error: -Os   scan-tree-dump-times original "(?n)f32 = __builtin_sqrtf \\(5\\.0e\\+0\\);$" 1
Error: -Os   scan-tree-dump-times original "(?n)f32 = __builtin_sinf \\(3\\.9e\\+1\\);$" 1
Error: -O2 -flto -fno-use-linker-plugin -flto-partition=none   scan-tree-dump-times original "(?n)f32 = __builtin_sqrtf \\(5\\.0e\\+0\\);$" 1
Error: -O2 -flto -fno-use-linker-plugin -flto-partition=none   scan-tree-dump-times original "(?n)f32 = __builtin_sinf \\(3\\.9e\\+1\\);$" 1
Error: -O2 -flto -fuse-linker-plugin -fno-fat-lto-objects   scan-tree-dump-times original "(?n)f32 = __builtin_sqrtf \\(5\\.0e\\+0\\);$" 1
Error: -O2 -flto -fuse-linker-plugin -fno-fat-lto-objects   scan-tree-dump-times original "(?n)f32 = __builtin_sinf \\(3\\.9e\\+1\\);$" 1
```

I expected to see this happen: passing

Instead, this happened: regression

### Meta

- What version of Rust GCC were you using, git sha if possible.

**Comment by philberty** [comment]
Did bors not pick this up?

**Comment by philberty** [comment]
this is was introduced in https://github.com/Rust-GCC/gccrs/pull/1630

**Comment by philberty** [comment]
fixed in #1676
